flink-runtime/src/test/java/org/apache/flink/runtime/execution/librarycache/BlobLibraryCacheRecoveryITCase.java

```diff
@@ -103,8 +107,8 @@ public void testRecoveryRegisterAndDownload() throws Exception {
 			cache = new PermanentBlobCache(config, blobStoreService, serverAddress[0]);
 
 			// Register uploaded libraries
-			ExecutionAttemptID executionId = new ExecutionAttemptID();
-			libServer[0].registerTask(jobId, executionId, keys, Collections.<URL>emptyList());
+			final LibraryCacheManager.ClassLoaderLease classLoaderLease = libServer[0].registerClassLoaderLease(jobId);
+			classLoaderLease.getOrResolveClassLoader(keys, Collections.emptyList());
```

Comment:
Are we doing this to download stuff into the `cache`?

Reply:
Yes exactly.